AIMS/Views/Home/AssetDetailsComponent.cshtml

```diff
@@ -1,141 +1,185 @@
-    var specsData = ViewData["SpecsData"] as Dictionary<string, Dictionary<string, string>>;
+  // --------- View data with safe fallbacks ----------
+  var category     = ViewData["Category"]?.ToString() ?? "Unknown";
+
+  // If you still receive these from the controller, that's fine; we won't rely on their order for headers.
+  var tableHeaders = ViewData["TableHeaders"] as List<string> ?? new List<string>();
 
-    ViewData["Title"] = $"{category} Asset Details";
+  var filteredData = ViewData["FilteredData"] as List<Dictionary<string, string>> ?? new();
+  var specsData    = ViewData["SpecsData"] as Dictionary<string, Dictionary<string, string>>
+                     ?? new Dictionary<string, Dictionary<string, string>>();
+
+  ViewData["Title"] = $"{category} Asset Details";
 }
+
 <script>
-    const specsData = @Html.Raw(Json.Serialize(ViewData["SpecsData"])); //gives access top specsData in JavaScript
+  // Expose specsData to JS (never null)
+  const specsData = @Html.Raw(JsonSerializer.Serialize(specsData));
 </script>
 
 <h2 class="text-center my-4">@ViewData["Title"]</h2>
 
-@if (filteredData != null && filteredData.Any())
+@if (filteredData.Any())
 {
-    <table class="asset-table">
-        <thead>
-            <tr>
-                @foreach (var header in tableHeaders)
-                {
-                    if (header == "Status")
-                    {
-                        <th class="d-flex justify-content-between align-items-center" style="gap: 10px;">
-    <span>Status</span>
-    <div style="margin-left: auto;">
-        @await Html.PartialAsync("_AssetButton")
-
+  <div class="asset-table-wrapper">
+    <!-- Right-aligned toolbar that shares the table width -->
+    <div class="asset-toolbar">
+      @await Html.PartialAsync("_AssetButton")
+      @* or your own button *@
+      @* <button type="button" class="btn btn-sm btn-primary">Manage Asset</button> *@
     </div>
-</th>
-
-                    }
-                    else
-                    {
-                        <th>@header</th>
-                    }
-                }
-            </tr>
-        </thead>
-        <tbody>
-            @foreach (var row in filteredData)
+
+    <table class="asset-table">
+      <colgroup>
+        <col style="width:30%" />
+        <col style="width:15%" />
+        <col style="width:12%" />
+        <col style="width:18%" />  @* Status *@
+        <col style="width:10%" />  @* Actions *@
+      </colgroup>
+
+      <thead>
+        <tr>
+          <th>Asset Name</th>
+          <th>Type</th>
+          <th>Tag #</th>
+          <th>
+            <div class="asset-header-bar">
+              <span>Status</span>
+            </div>
+          </th>
+          <th class="actions-col">Actions</th>
+        </tr>
+      </thead>
+
+      <tbody>
+        @foreach (var row in filteredData)
+        {
+            var assetName = row.TryGetValue("Asset Name", out var _n) ? _n : "";
```

Comment:
As a question for the general team: When/if this data fails to load, do we want to show an empty string or instead present the user with some error?